File: LogicadeProgramacao/Estrutura de Decisão/3. Python/EquacaoSegundoGrau.py
a = float(input("Digite o valor de a: "))
if a == 0:
    print("O valor de a deve ser diferente de zero para que a equação seja do segundo grau.")
else:
    b = float(input("Digite o valor de b: "))
    c = float(input("Digite o valor de c: "))
    delta = b ** 2 - 4 * a * c

if delta < 0:
    print("A equação não possui raízes reais.")
elif delta == 0:
    raiz = -b / (2 * a)
    print("A equação possui apenas uma raiz real: ", raiz)
else:
    raiz1 = (-b + delta ** 0.5) / (2 * a)
    raiz2 = (-b - delta ** 0.5) / (2 * a)
    print("A equação possui duas raízes reais:")
    print("Raiz 1 = ", raiz1)
    print("Raiz 2 = ", raiz2)   

File: LogicadeProgramacao/Estrutura de Decisão/2. Java/EquacaoSegundoGrau.java
import java.util.Scanner;

public class EquacaoSegundoGrau {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        double a, b, c, delta, x1, x2;

        System.out.print("Digite o valor de A: ");
        a = sc.nextDouble();

        if (a == 0) {
            System.out.println("Não é uma equação de segundo grau!");
        } else {
            System.out.print("Digite o valor de B: ");
            b = sc.nextDouble();
            System.out.print("Digite o valor de C: ");
            c = sc.nextDouble();

            delta = b * b - 4 * a * c;

            if (delta < 0) {
                System.out.println("A equação não possui raízes reais!");
            } else if (delta == 0) {
                x1 = (-b) / (2 * a);
                System.out.println("A equação possui apenas uma raiz real: " + x1);
            } else {
                x1 = (-b + Math.sqrt(delta)) / (2 * a);
                x2 = (-b - Math.sqrt(delta)) / (2 * a);
                System.out.println("A equação possui duas raízes reais: " + x1 + " e " + x2);
            }
        }

        sc.close();
    }
}